### PR TITLE
refactor: Improve error when submitting block

### PR DESCRIPTION
**Pull request by doitian**
Include more info in the error messages:

* Add expected version and actual value in header version error.

Add logs to ease debugging PoW. It can be enabled by setting the log level:

```
[logger]
filter = "info,ckb-pow=debug"
```

### CODE REVIEW

**Comment by doitian** [comment]
Rebased to resolve conflicts in verification/src/error.rs

**Comment by quake** [comment]
bors r=quake,yangby-cryptape

File: Cargo.lock [hunk 1]
```diff
@@ -775,6 +775,7 @@ dependencies = [
  "ckb-hash",
  "ckb-types",
  "eaglesong",
+ "log 0.4.8",
```
Comment (yangby-cryptape) [inline Cargo.lock:778]: Linters Error: `Cargo.lock` is changed after build.
```suggestion
 "log 0.4.11",
```

Comment (doitian) [reply Cargo.lock:778]: Roger

**Comment by doitian** [comment]
Resolved the Cargo.lock conflict

**Comment by doitian** [comment]
bors r=quake,keroro520,yangby-cryptape

**Pull request merged**